src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -393,10 +399,11 @@ public Response delete(@Auth AuthUser authUser, @PathParam("datasetId") Integer
   @POST
   @Path("/index")
   @RolesAllowed(ADMIN)
-  public Response indexDatasets() {
+  public Response indexDatasets(@Auth AuthUser authUser) {
     try {
+      User user = userService.findUserByEmail(authUser.getEmail());
```

Comment:
Something for a future PR; I think you can create an `Authenticator` to support injecting `User` directly. Conceptually it would inject `AuthUser` and then call `userService.findUserByEmail()` like this code does.